src/Functions/array/arrayZip.cpp

```diff
@@ -171,21 +171,27 @@ class FunctionArrayZip : public IFunction
-)",
-         .category = FunctionDocumentation::Category::Array});
-
-    factory.registerFunction<FunctionArrayZip<true>>(
-        {.description = R"(
-Combines multiple arrays into a single array, allowing for unaligned arrays. The resulting array contains the corresponding elements of the source arrays grouped into tuples in the listed order of arguments.
-
-If the arrays have different sizes, the shorter arrays will be padded with `null` values.
-)",
-         .category = FunctionDocumentation::Category::Array}
-
-    );
+    FunctionDocumentation::Description description = "Combines multiple arrays into a single array. The resulting array contains the corresponding elements of the source arrays grouped into tuples in the listed order of arguments.";
+    FunctionDocumentation::Syntax syntax = "arrayZip(arr1, arr2, ..., arrN)";
+    FunctionDocumentation::Arguments argument = {{"arr1, arr2, ..., arrN", "N arrays to combine into a single array. [`Array(T)`](/sql-reference/data-types/array)"}};
+    FunctionDocumentation::ReturnedValue returned_value = "Returns an array with elements from the source arrays grouped in tuples. Data types in the tuple are the same as types of the input arrays and in the same order as arrays are passed. [`Array(T)`](/sql-reference/data-types/array)([`Tuple`](/sql-reference/data-types/tuple)).";
+    FunctionDocumentation::Examples example = {{"Usage example", "SELECT arrayZip(['a', 'b', 'c'], [5, 2, 1]);", "[('a',5),('b',2),('c',1)]"}};
+    FunctionDocumentation::IntroducedIn introduced_in = {20, 1};
+    FunctionDocumentation::Category category = FunctionDocumentation::Category::Array;
+    FunctionDocumentation documentation = {description, syntax, argument, returned_value, example, introduced_in, category};
+
+    factory.registerFunction<FunctionArrayZip<false>>(documentation);
+
+    FunctionDocumentation::Description description_unaligned = "Combines multiple arrays into a single array, allowing for unaligned arrays (arrays of differing lengths). The resulting array contains the corresponding elements of the source arrays grouped into tuples in the listed order of arguments.";
+    FunctionDocumentation::Syntax syntax_unaligned = "arrayZipUnaligned(arr1, arr2, ..., arrN)";
+    FunctionDocumentation::Arguments argument_unaligned = {{"arr1, arr2, ..., arrN", "N arrays to combine into a single array. [`Array(T)`](/sql-reference/data-types/array)."}};
+    FunctionDocumentation::ReturnedValue returned_value_unaligned = "Returns an array with elements from the source arrays grouped in tuples. Data types in the tuple are the same as types of the input arrays and in the same order as arrays are passed. [`Array(T)`](/sql-reference/data-types/array)([`Tuple(T1, T2, ...)`](/sql-reference/data-types/tuple)).";
+    FunctionDocumentation::Examples example_unaligned = {{"Usage example", "SELECT arrayZipUnaligned(['a'], [1, 2, 3]);", "[('a',1),(NULL,2),(NULL,3)]"}};
```

Comment:
Cosmetics: Let's use whitespace after comma consistently (`, ` is better than `,`)

Reply:
The result is returned without a space after a comma. We should have a setting for that! I propose a new setting `output_format_space_after_comma_in_query_result`. There are not enough settings in ClickHouse, we could use another. :D